src/markdownRenderer.ts

```diff
@@ -0,0 +1,502 @@
+    )
+
+    // Add checks that failed
+    if (checks && checks.length > 0) {
+      // Aggregate checks by name
+      const checksByName: Record<
+        string,
+        { success_count: number; fail_count: number; total_count: number }
+      > = {}
+
+      // Group checks by name and aggregate metrics
+      checks.forEach((check) => {
+        const { name, metric_summary } = check
+        if (!checksByName[name]) {
+          checksByName[name] = {
+            success_count: 0,
+            fail_count: 0,
+            total_count: 0,
+          }
+        }
+        checksByName[name].success_count += metric_summary.success_count
+        checksByName[name].fail_count += metric_summary.fail_count
+        checksByName[name].total_count +=
+          metric_summary.success_count + metric_summary.fail_count
+      })
+      // List failed checks (those with fail_count > 0)
+      console.log(`Checks by name`)
+      console.log(JSON.stringify(checksByName, null, 2))
+      Object.entries(checksByName)
+        .filter(([, metrics]) => metrics.fail_count > 0)
+        .forEach(([name, metrics]) => {
+          markdownSections.push(
+            `  - \`${name}\`: Failed **${formatNumber(metrics.fail_count)}**, out of **${formatNumber(metrics.total_count)}** times.`
+          )
+        })
+    }
+  } else {
+    markdownSections.push(
+      `- ✅ All **${formatNumber(checksMetrics.total)}** checks were successful.`
+    )
+  }
+
+  return markdownSections
+}
+
+/**
+ * Generates markdown for thresholds metrics
+ * @param thresholdsMetrics Thresholds metrics data
+ * @returns Markdown string for thresholds metrics
+ */
+export function getThresholdsMarkdown(
+  thresholdsMetrics: ThresholdsSummary | null
+): string[] {
+  if (
+    !thresholdsMetrics ||
+    Object.keys(thresholdsMetrics).length === 0 ||
+    thresholdsMetrics.total == null ||
+    thresholdsMetrics.total <= 0
+  ) {
+    return []
+  }
+
+  const markdownSections = []
+
+  if (
+    thresholdsMetrics.successes != null &&
+    thresholdsMetrics.successes < thresholdsMetrics.total
+  ) {
+    markdownSections.push(
+      `- ❌ **${formatNumber(thresholdsMetrics.total - thresholdsMetrics.successes)}** out of **${formatNumber(thresholdsMetrics.total)}** thresholds were not met.`
+    )
+  } else {
+    markdownSections.push(
+      `- ✅ All **${formatNumber(thresholdsMetrics.total)}** thresholds were met.`
+    )
+  }
+
+  return markdownSections
+}
+
+/**
+ * Returns a markdown string representing the test run status
+ * @param testRunStatus The status code of the test run
+ * @returns Markdown string for test run status
+ */
+export function getTestRunStatusMarkdown(
+  testRunStatus: number | undefined
+): string {
+  let statusString = ''
+
+  if (testRunStatus === undefined || testRunStatus === null) {
+    statusString = '❓ Unknown'
+  } else if (testRunStatus === 3) {
+    statusString = '✅ Passed'
```

Comment:
Consider using a combination of the run and result status for better results. This code will mark the test as passed even if some of the thresholds are tainted and the result status is "Failed".